rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,771 @@
+
+- selector `.fzbuleu > *`
+- match_attempts **502**
+- match_count **1**
+
+##### `> h1` & `> div` (non ideal)
+
+Targets all `h1` & `div` tags.
+
+- selector `.fohk16y > h1`
+- match_attempts **1**
+- match_count **1**
+
+In this example performs better as page has a single `h1` tag, this will not be a case in the real app that it is visible on the next example with `div`.
+
+- selector `.fq4d7o6 > div`
+- match_attempts **501**
+- match_count **1**
+
+#### Targeting a classname (better)
+
+The most effective out there as we target only elements with a specific classname. However, it will not save from the case with 1k buttons in the app.
+
+- selector `.fqhvij7 > .ui-button`
+- match_attempts **1**
+- match_count **1**
+
+# Best practices
+
+## Writting styles
+
+### Use `tokens` over direct colors
+
+Fluent UI React v9 provides [design tokens][fluent-colors] to provide consistent way of theming via `tokens` provided by `@fluentui/react-theme`.
+
+```js
+import { makeStyles } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  // ❌ Don't do
+  rootA: { color: 'red' /* brand foreground */ },
+  // ✅ Do
+  rootB: { color: tokens.colorBrandForeground1 },
+});
+```
+
+Exact colors should be never used in Fluent UI code and are not recommended by be used in applications. The exception are [system-colors][mdn-system-colors] used by forced colors mode:
+
+```js
+import { makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  button: { color: 'ButtonText' },
+});
+```
+
+### Avoid duplication in rule definitions
+
+Styles written for components should follow these rules:
+
+- base styles should contain the most of CSS definitions that are applicable to base state
+- styles for permutations should be granular and don't contain definitions that are present in base styles
+
+```js
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  base: {
+    display: 'flex',
+    color: tokens.colorNeutralForeground1,
+    ...shorthands.padding('10px'),
+  },
+  // ❌ Don't do
+  //  "display" & "padding" with the same values are defined in base styles
+  primary: {
+    display: 'flex',
+    ...shorthands.padding('10px'),
+    backgroundColor: tokens.colorBrandBackground,
+    color: tokens.colorBrandForeground1,
+  },
+  // ✅ Do
+  primary: {
+    backgroundColor: tokens.colorBrandBackground,
+    color: tokens.colorBrandForeground1,
+  },
+});
+
+function App(props) {
+  const classes = useClasses();
+  const className = mergeClasses(classes.base, props.primary && classes.primary);
+
+  /* --- */
+}
+```
+
+### Avoid `!important`
+
+Our styles are written in way to allow predictable and simple style overrides, `!important` should not be used in styles:
+
+```js
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+
+const useClasses = makeStyles({
```

Comment:
This example could be better, maybe let's do an example of how overrides work (i.e. based on application order... perhaps this could be in `basics` section too